[16] 달력 내부 투두 UI

soozynn opened the pull request:
# [16]  달력 & 내부 todo
## 노션 칸반 링크
- [[Task] 달력 & 내부 todo](https://www.notion.so/vanillacoding/Task-todo-0b3e7a56d45448bea2a316ab61ae34c7)

## 카드에서 구현 혹은 해결하려는 내용
- 날짜와 일치하는 todo를 달력에 순서대로 추가 
- 달력 UI


## 테스트 방법
- login 시 달력에 추가한 todo가 있을 경우, 달력에 그려짐

## 기타 사항
- [현재 캘린더 todo 모달](https://www.notion.so/vanillacoding/Task-todo-task-16-ef8607a34ff3494cb02c409af4b9ae21)과 같이 진행 중입니다. 두 가지 태스크를 합쳐서 같이 올리려다가 시간이 지체될 것 같아 UI 태스크 먼저 올립니다. 
- 모달도 오늘 안으로 추가적으로 올릴 계획인데 실데이터를 기반으로 연동해보고 데이터를 가져오는 부분에서 시간이 조금 소요가 되고 있습니다. 최대한 빨리 올려보도록 하겠습니다.🙏

Commit:
[16] 달력 내부 투두 UI

- 달력 내의 todo 추가 구현
- 달력 UI

lazymole commented on src/components/CalendarPage.js:
```diff
@@ -43,45 +65,59 @@ export default function CalendarPage() {
+        <div className="date">{format(currentDate, "yyyy년 MM월")}</div>
+        <div onClick={handleNextButtonClick} className="week-button">{`>`}</div>
+      </Calendar>
+      <Calendar className="calender">{showWeekCalendar()}</Calendar>
```
```suggestion
      <Calendar className="calendar">{showWeekCalendar()}</Calendar>
```

lazymole commented on src/components/Day.js:
```diff
@@ -0,0 +1,61 @@
+import Todo from "./Todo";
+
+const DayContainer = styled.div`
+  font-family: "Acme", sans-serif;
+  font-family: "Caveat", cursive;
+  font-family: "Gochi Hand", cursive;
+  font-family: "Quintessential", cursive;
```
font-family는 순서대로 적용되는 것이라, 한 번만 선언해줘도 되리라 봅니다.
그나저나 소나무 취향이시군요. 🤔

```suggestion
  font-family: Acme, Caveat, "Gochi Hand", Quintessential, sans-serif;
```

lazymole commented on src/components/Day.js:
```diff
@@ -0,0 +1,61 @@
+  font-family: "Hanna", sans-serif;
+  padding: 10px 0;
+  background: white;
+  background-color: rgb(148, 178, 235);
```
이 부분도 색상만 지정하실 거라면 둘 중 하나만 쓰시면 될 것 같네요! 저 개인적으로는 색상은 rgb나 Hex Code를 선호하는 편이에요.

lazymole commented on src/components/LoginPage.js:
```diff
@@ -11,15 +11,22 @@ const LoginModal = styled.div`
   border-radius: 15px;
   text-align: center;
 
+  h1 {
+    font-family: "Acme", sans-serif;
+    font-family: "Caveat", cursive;
+    font-family: "Gochi Hand", cursive;
+    font-family: "Quintessential", cursive;
+  }
```
먼저 드린 코멘트 참고해주세요!

lazymole commented on src/components/LoginPage.js:
```diff
@@ -88,14 +96,13 @@ export default function LoginPage() {
         <LoginContainer>
           <LoginModal>
-            <img className="logo" alt="logo" src="../img/logo.png" />
+            <img className="logo" alt="logo" src={"/img/logo.png"} />
```
중괄호 없어도 되지 않을까요!

```suggestion
            <img className="logo" alt="logo" src="/img/logo.png" />
```

lazymole commented on src/components/Todo.js:
```diff
@@ -0,0 +1,108 @@
+  const [showModal, setShowModal] = useState(false);
+
+  const handleTodoClick = () => {
+    console.log(todos, date);
```
```suggestion
```

lazymole commented on src/components/Todo.js:
```diff
@@ -0,0 +1,108 @@
+            )}
+            {isComplete ? (
+              <Title className="complete" onClick={handleTodoClick}>
+                {"전수진"}
```
전용 달력? 👀

lhc0506 commented on src/components/CalendarPage.js:
```diff
@@ -43,45 +65,59 @@ export default function CalendarPage() {
   useEffect(() => {
     if (loginState) {
-      getTodosOfWeek();
+      getUsersTodo();
```
프론트단 합쳐서, login못하면 페이지에 들어올 수 없어서 105번 조건은 이제 빼주셔도 될것같습니다!

lhc0506 commented on src/components/Todo.js:
```diff
@@ -0,0 +1,108 @@
+    setShowModal(true);
+  };
+
+  const handleCheckButtonClick = event => {
+    setIsComplete(false);
+    event.stopPropagation();
+  };
+
+  const handleNounCheckButtonClick = event => {
+    setIsComplete(true);
+    event.stopPropagation();
+  };
+
+  return (
+    <>
+      {todos.forEach(todo => {
+        return (
+          <TodoContainer>
+            {isComplete ? (
+              <CheckButton
+                src={"/img/checkButton.png"}
+                onClick={handleCheckButtonClick}
+              />
+            ) : (
+              <NounCheckButton
+                src={"/img/nounCheck.png"}
+                onClick={handleNounCheckButtonClick}
+              />
+            )}
```
handleClick과 Button  Styled tag를 하나로 줄일 수 있을 것 같아요~
```
const handleCheckButtonClick = event => {
  setIsComplete(!isComplete);
  event.stopPropagation();
}

<CheckButton
  src{isComplete ? "/img/checkButton.png" : "/img/nounCheck.png"}
  onClick={handleCheckButtonClick}
/>
```

lhc0506 commented on src/components/Todo.js:
```diff
@@ -0,0 +1,108 @@
+                {"전수진"}
+              </Title>
+            ) : (
+              <Title onClick={handleTodoClick}>{"전수진"}</Title>
```
조건연산자를 `className={isComplete ? "complete" : null}`으로 줄 수 있지 않을까요??
혹은 `isComplete && "complete"` 로요

soozynn replied:
그렇겠네요! 해당 부분 수정하도록 하겠습니다

Commit:
[16] 달력 내부 투두 UI

- Feat: handleButtonClick 으로 중복 로직 최소화
- Feat: 삼항연산자를 className 내부에서 작성

[16] 달력 내부 투두 UI

- Feat: handleCheckButtonClick으로 중복 로직 최소화
- Feat: 삼항연산자를 className 내부에서 조건 선언
